components/filterModal/detailFilterModal.tsx

```diff
@@ -0,0 +1,175 @@
+'use client';
+
+import { ChangeEvent, useState } from 'react';
+import DatePicker from 'react-datepicker';
+
+import Image from 'next/image';
+
+import 'react-datepicker/dist/react-datepicker.css';
+import Button from '../button/button';
+
+const ADDRESS_LIST = [
+  '서울시 종로구',
+  '서울시 중구',
+  '서울시 용산구',
+  '서울시 성동구',
+  '서울시 광진구',
+  '서울시 동대문구',
+  '서울시 중랑구',
+  '서울시 성북구',
+  '서울시 강북구',
+  '서울시 도봉구',
+  '서울시 노원구',
+  '서울시 은평구',
+  '서울시 서대문구',
+  '서울시 마포구',
+  '서울시 양천구',
+  '서울시 강서구',
+  '서울시 구로구',
+  '서울시 금천구',
+  '서울시 영등포구',
+  '서울시 동작구',
+  '서울시 관악구',
+  '서울시 서초구',
+  '서울시 강남구',
+  '서울시 송파구',
+  '서울시 강동구',
+];
+// Todo: key값 수정, 적용하기 버튼 onClick 이벤트 추가하기, ESlint 들여쓰기 이슈 고치기
+function DetailFilterModal() {
+  const [activeModal, setActiveModal] = useState<boolean>(false);
+  const [startDate, setStartDate] = useState<Date | null>(null);
+  const [clickAddressList, setClickAddressList] = useState<string[]>([]);
+  const [filterAddressList, setFilterAddressList] = useState<string[]>([]);
+  const [searchValue, setSearchValue] = useState<string>('');
+  const [amount, setAmount] = useState<number | null>();
+  const handleAddClickAddressListItem = (address: string) => {
+    if (clickAddressList.indexOf(address) !== -1) return;
+    const addressAddArr = [...clickAddressList, address];
+    setClickAddressList(addressAddArr);
+  };
+  const handleDeleteClickAddressListItem = (address: string) => {
+    const sliceArr = [...clickAddressList];
+    const addressIndex = clickAddressList.indexOf(address);
+    sliceArr.splice(addressIndex, 1);
+    setClickAddressList(sliceArr);
+  };
+  const handleChangeAmount = (e: ChangeEvent<HTMLInputElement>) => {
+    setAmount(+e.target.value);
+  };
+  const handleSearchAddress = (e: ChangeEvent<HTMLInputElement>) => {
+    setSearchValue(e.target.value);
+    const reg = new RegExp(`${searchValue}`);
+    const filterArr = ADDRESS_LIST.filter((address) => reg.test(address));
+    setFilterAddressList(filterArr);
+  };
```

Comment:
P4: `includes` method를 활용해보는 것도 좋을 것 같습니다. 또한, 코드를 돌려보지는 않아서 될지는 모르겠지만, `FilterAddressList`의 값을 state를 사용하지 않고 받아와 볼 수 있을 것 같습니다...!

```suggestion
const handleSearchAddress = useCallback((e: ChangeEvent<HTMLInputElement>) => {
  const value = e.target.value;
  setSearchValue(value);
}, []);

const filteredAddressList = ADDRESS_LIST.filter(address => address.includes(searchValue));
```